protobuf/endpoints/batch_create_artifact.proto

```diff
@@ -2,9 +2,18 @@ syntax = "proto3";
 
 package endpoints;
 
-// This config is included for potential future use.
-// In the future it may include part count, expiration, or other metadata.
+enum ArtifactType {
+  ARTIFACT_TYPE_UNSPECIFIED = 0;
+  ARTIFACT_TYPE_PARAMS = 1;
+  ARTIFACT_TYPE_RESULT = 2;
+}
+
 message CreateArtifactConfig {
+  // Required field representing the type of content the artifact will contain.
+  optional ArtifactType type = 1;
+
+  // Optional field representing the initial number of parts to allocate for this artifact.
```

Comment:
or should this field be required?

Reply:
I thought required was toxic

Reply:
I don't mean literally. `proto3` doesn't even have `required`. I mean semantically, like according to our documentation

Reply:
Oh. I don't think we need to